parser-typechecker/src/Unison/Codebase/SqliteCodebase/Conversions.hs

```diff
@@ -559,3 +560,27 @@ branch2to1 branchCache lookupCT (V2.Branch.Branch v2terms v2types v2patches v2ch
               vals :: Relation.Relation ref (V1.Metadata.Type, V1.Metadata.Value) =
                 Relation.insertManyRan ref (fmap (\(v, t) -> (mdref2to1 t, mdref2to1 v)) (Map.toList mdvals)) mempty
            in star <> V1.Star3.Star3 facts names types vals
+
+-- | Generates a v1 short hash from a v2 referent.
+-- Also shortens the hash to the provided length. If 'Nothing', it will include the full
+-- length hash.
+referent2toshorthash1 :: Maybe Int -> V2.Referent -> V1.ShortHash.ShortHash
```

Comment:
I'm still mulling over that this direction shouldn't happen except perhaps in a display layer, which could then skip `ShortHash` altogether.

Reply:
It doesn't have to be in this PR, but wdyt

Reply:
I would love if that were the case, but it begs the question of whether our API layer should need to have a handle on the codebase. given Referent came from in order to compute the shorthash, I think the answer to that should be no, and that we should use a global constant for the length of shorthashes.

What do you think of that?